### ISSUE TITLE
Balance negatives traits "Weight:XS" and"Weight:XXXL"

### ISSUE THREAD

**Issue by Proxima-b** [CleverRaven/Cataclysm-DDA#53922]
### Is your feature request related to a problem? Please describe.

Negatives traits "Weight:XS" and"Weight:XXXL" both cost -1 trait point which means they should be two negative traits with similar severity.

However, my in-game experience suggests otherwise. In fact, XS trait is wayyyyyyyyyyyyyyy more negative sides than XXXL trait.

For starters, XXXL body weight means the character won't have any foraging need in the first couple of weeks, they only require water, which ease the playthru quite a lot.  Meanwhile, XS body weight means the character will soon die of starvation without a reliable food source early-game, which could be a huge pain in the ass.

What's more ridiculous is that XS body weight carries all sorts of debuffs to the characters, such as severe decrease in all stats and speed. While XXXL body weight has almost no drawbacks other than the negative health modifier, which can be resolved simply by fasting.

I will admit having more fat could be beneficial in the post-apocalyptic world. That is why given current trait behaviour, "weight XXXL" should be a 0 point trait, and "weight XS" should cost -2 points. Some adjsutments to weight behaviour would also be acceptable.

### Solution you would like.

Set weight:XXXL trait cost at 0
Set weight:XS trait cost at -2

### Describe alternatives you have considered.

Set weight:XXXL trait cost at -1
Set weight:XS trait cost at -2
Add more penalties for going overweight and obesity, those conditions should not only affect health, but also have significant drawbacks on agility and speed.

**Comment by PatrikLundell** [comment]
I would prefer to add more negative effects to being obese and above. As it currently stands, XXXL is an easy pick with the drawback being a more or less guaranteed cold early on. At a cost of 0, however, I see no reason to take it except for role playing purposes.

I wouldn't want to introduce penalties for overweight unless you split it into mildly overweight without penalties and overweight with penalties, because unless you're submitting yourself to slavery under the bathroom scale (with the attendant handling of equipment weight), there's no way to control your weight to always stay within the non penalized zone. Currently you can eat until you hit overweight, scale back a notch, eat more when you get the calorie deficiency warning, increase the eating, eventually hit overweight, scale back...

**Comment by anoobindisguise** [comment]
Being fat should be a lot more penalizing for sure. Between clothes being (poor fit), encumbrance, lack of stamina, and also no longer being able to just undergo the "Water and Multivitamins" diet to lose weight there are a lot of serious considerations to manage that are not represented at all.

**Comment by estebandellasilva** [comment]
> Being fat should be a lot more penalizing for sure. Between clothes being (poor fit), encumbrance, lack of stamina, and also no longer being able to just undergo the "Water and Multivitamins" diet to lose weight there are a lot of serious considerations to manage that are not represented at all.

Slowing down the Speed could be an Option being fat means most of the times you your reaction speed iss limited to you fat limbs moving at a slower speed

**Comment by NetSysFire** [comment]
I could imagine there being debuffs for going up the stairs and changing the movement mode. Stamina should run out much more quickly, making you extremely vulnerable. If you just camp out in the woods, this should not affect you much, about according to https://en.wikipedia.org/wiki/Obesity#Effects_on_health, pain is also a thing. Maybe a debuff triggered by certain action can cause random back pain or similar. Distracting pain seems to be the most appropriate pain level here.
Since it can also cause sleep apnea, reduced sleep quality might also be good to add to that.

NPCs who respect the pretty trait could also react negatively to the characters weight. Since there can also be more complications when the character is under anesthesia, there could be an increased failure chance with autodocs perhaps.

This is all passive. You need to be active when trying to get rid of being severely underweight, so there should also be some sort of requirement when trying to lose much weight.

**Comment by Proxima-b** [comment]
> I could imagine there being debuffs for going up the stairs and changing the movement mode. Stamina should run out much more quickly, making you extremely vulnerable. If you just camp out in the woods, this should not affect you much, about according to https://en.wikipedia.org/wiki/Obesity#Effects_on_health, pain is also a thing. Maybe a debuff triggered by certain action can cause random back pain or similar. Distracting pain seems to be the most appropriate pain level here. Since it can also cause sleep apnea, reduced sleep quality might also be good to add to that.
> 
> NPCs who respect the pretty trait could also react negatively to the characters weight. Since there can also be more complications when the character is under anesthesia, there could be an increased failure chance with autodocs perhaps.
> 
> This is all passive. You need to be active when trying to get rid of being severely underweight, so there should also be some sort of requirement when trying to lose much weight.

Agreed, I would love to right some codes for such changes, however I cannot find the file which represents body weight behaviour.

**Comment by NetSysFire** [comment]
Theres a couple it seems, but I am no C++ coder:

https://github.com/CleverRaven/Cataclysm-DDA/blob/1ea2a93461635532285c43b00bb9adf1e1190a31/src/character.cpp#L4749-L4782

https://github.com/CleverRaven/Cataclysm-DDA/blob/c6e025aba8f40a12dfbed9c1404be30e302d680b/src/character_body.cpp#L846-L909

https://github.com/CleverRaven/Cataclysm-DDA/blob/427c1c682518ba7839111bf097a65de4009d0de1/src/player_display.cpp#L1213-L1235

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/obesity-is-extemely-poorly-balanced-and-shoud-be-fixed/27919/2

**Comment by anoobindisguise** [comment]
This can be closed owing to both my changes to the BMI system and the soft removal of points.
